src/module_wrap.cc

```diff
@@ -494,6 +493,59 @@ Maybe<uv_file> CheckFile(const URL& search,
   return Just(fd);
 }
 
+PackageConfig emptyPackage = { false, false, "" };
+std::unordered_map<std::string, PackageConfig> pjson_cache_;
+
+PackageConfig& GetPackageConfig(Environment* env, const std::string path) {
+  auto existing = pjson_cache_.find(path);
+  if (existing != pjson_cache_.end()) {
+    return existing->second;
+  }
+  Maybe<uv_file> check = CheckFile(path, LEAVE_OPEN_AFTER_CHECK);
+  if (check.IsNothing()) {
+    return pjson_cache_[path] =
+           emptyPackage;
+  }
+
+  Isolate* isolate = env->isolate();
+  Local<Context> context = isolate->GetCurrentContext();
+  std::string pkg_src = ReadFile(check.FromJust());
+  uv_fs_t fs_req;
+  uv_fs_close(nullptr, &fs_req, check.FromJust(), nullptr);
+  uv_fs_req_cleanup(&fs_req);
+
+  // It's not okay for the called of this method to not be able to tell
+  // whether an exception is pending or not.
+  TryCatch try_catch(isolate);
+
+  Local<String> src;
+  if (!String::NewFromUtf8(isolate,
+                           pkg_src.c_str(),
+                           v8::NewStringType::kNormal,
+                           pkg_src.length()).ToLocal(&src)) {
+    return pjson_cache_[path] =
+           emptyPackage;
+  }
+
+  Local<Value> pkg_json;
+  if (!JSON::Parse(context, src).ToLocal(&pkg_json) || !pkg_json->IsObject())
```

Comment:
Can you use the ToObject() overload that takes a Context?

Reply:
I'm' not sure I follow exactly what is intended here. Do you mean something like:

```js
Local<Value> pkg_json = JSON::Parse(context, src).ToLocalChecked();
MaybeLocal<Object> pkg_json_obj = pkg_json->ToObject(context);
if (pkg_json_obj.isEmpty())
// ...
```

Reply:
You probably don't want to use `.ToLocalChecked()` on the return value of `JSON::Parse()` because that will abort node when the JSON is malformed without a clear error message (aside: that's also an argument for removing the TryCatch; the exception should bubble up.)

Essentially, you'd do something like this:
```c++
Local<Value> package_json_v;
Local<Object> package_json;
if (!JSON::Parse(context, src).ToLocal(&package_json_v)) return;  // Exception pending.
if (!package_json_v->ToObject(context).ToLocal(&package_json)) return;  // Exception pending.
// |package_json| points to a valid JS object now.
```